lib/matplotlib/cbook/deprecation.py

```diff
@@ -329,10 +329,12 @@ def func(used_arg, other_arg, unused, more_args): ...
     """
 
     if func is None:
-        return functools.partial(_delete_parameter, since, name)
+        return functools.partial(_delete_parameter, since, name, **kwargs)
 
     signature = inspect.signature(func)
-    assert name in signature.parameters, (
+    kw_name = next((param.name for param in signature.parameters.values()
```

Comment:
```suggestion
    # name of kwargs parameter, e.g. 'kwargs' if **kwargs is in the signature of the
    # decorated function, or None if the function does not have such a parameter
    kwargs_name = next((param.name for param in signature.parameters.values()
```
Even though this may seem picky, this is rather unusual code and every bit of clarity helps (connection `kwargs <-> kwargs_name`).

Reply:
done